src/main/java/mate/academy/spring/controller/ShoppingCartController.java

```diff
@@ -0,0 +1,44 @@
+package mate.academy.spring.controller;
+
+import mate.academy.spring.mapper.impl.response.ShoppingCartResponseMapper;
+import mate.academy.spring.model.dto.response.ShoppingCartResponseDto;
+import mate.academy.spring.service.MovieSessionService;
+import mate.academy.spring.service.ShoppingCartService;
+import mate.academy.spring.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/shopping-carts")
+public class ShoppingCartController {
+    private final UserService userService;
+    private final ShoppingCartService shoppingCartService;
+    private final ShoppingCartResponseMapper shoppingCartResponseMapper;
```

Comment:
```suggestion
    private final DtoResponseMapper<ShoppingCartResponseDto, ShoppingCart> shoppingCartDtoResponseMapper;
```
You must depend from abstraction, not from implementation (see check_list)

Reply:
ok